server/sonar-server/src/main/java/org/sonar/server/permission/UserPermissionChanger.java

```diff
@@ -58,6 +62,9 @@ public boolean apply(DbSession dbSession, UserPermissionChange change) {
       default:
         throw new UnsupportedOperationException("Unsupported permission change: " + change.getOperation());
     }
+    if (SYSTEM_ADMIN.equals(change.getPermission())) {
+      dbClient.userDao().updateRootFlagFromPermissions(dbSession, change.getUserId().getLogin(), defaultOrganizationProvider.get().getUuid());
```

Comment:
the root flag relates to global permissions only. You can ignore modifications of project permissions by adding the condition on `!change.getProjectId().isPresent()`.